### PR TITLE
VAMC-10097 Add Facility Health Service Migration

### PR DESCRIPTION
**Pull request by sundarka1**
## Description

Load Facility Health Services data from CSV

## QA steps

As admin 

Validate the node "Ophthalmology at VA Wilkes-Barre health care - Allentown VA Clinic"

Node data

Screen Shot 2020-07-15 at 1 34 48 PM

- [x] Validate Lookup gets correct "Facility".

- [x] Validate Lookup gets correct "VAMC system health service".

- [x] Validate WYSIWIG content looks correct(Facility description of service).

- [x] Validate phone number links in wysiwyg are valid html that result in a tel dialer.
 
- [x] Validate Text format set to "Rich text"
 
- [x] Validate  draft time, author set correctly.
 
- [x] Validate "Owner" set correctly.
 
Revisions data

Screen Shot 2020-07-15 at 1 34 48 PM

- [x] Validate revision date and author

- [x] Validate  Log message meaningful (Imported facility health services)

- [x] Validate Check owner is set for "VISN 4 > VA Wilkes-Barre health care"
 
- [x] Validate Check Recent change log information matches.

### CODE REVIEW

File: docroot/modules/custom/va_gov_migrate/data/VAMC_Facility_Health_Services_Wilkes-Barre.csv [hunk 1]
```diff
@@ -0,0 +1,8 @@
+Id;Facility;VAMC System Health Service;Facility description of services;Owner
```
Comment (swirtSJW) [inline docroot/modules/custom/va_gov_migrate/data/VAMC_Facility_Health_Services_Wilkes-Barre.csv:1]: Make filename more generic to cover the content type not the specific system.

File: docroot/modules/custom/va_gov_migrate/data/VAMC_Facility_Health_Services_Wilkes-Barre.csv [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+'VAMC-WB-7';'Allentown VA Clinic';'Ophthalmology at VA Wilkes-Barre health care';'<h3>Location and contact information</h3><p class=""va-address-block""><strong></strong><br />3110 Hamilton Boulevard, <br />Allentown, PA 18103-3630<br /><strong>Phone:</strong> <a href=""tel: +1610-776-4304"">610-776-4304</a><br /><strong>Hours:</strong> Monday: 8:00 a.m. to 8:00 p.m. ET, <br />Tuesday through Friday: 8:00 a.m. to 4:30 a.m. ET</p><h3>Appointments</h3><p>Contact us to schedule, reschedule, or cancel your appointment.  If a referral is required, you will need to contact your primary care provider first.</p><p><strong>Referral needed? </strong> N<br /><strong>Walk-ins accepted?</strong> N<br /><strong>Phone: </strong><a href=""tel:+1610-776-4304"">610-776-4304</a> </p>';'VA Wilkes-Barre health care'
```
Comment (swirtSJW) [inline docroot/modules/custom/va_gov_migrate/data/VAMC_Facility_Health_Services_Wilkes-Barre.csv:8]: Do we have more data than this?  I don't want to repeat the issues of the previous migration where the data kept shifting and breaking the migration.

Comment (sundarka1) [reply docroot/modules/custom/va_gov_migrate/data/VAMC_Facility_Health_Services_Wilkes-Barre.csv:8]: Okay, I'll process with all the facilities.

File: config/sync/migrate_plus.migration.facility_health_care_service.yml [hunk 3]
```diff
@@ -0,0 +1,87 @@
+uuid: d6f0d43f-2e80-4224-9013-982136c095c8
+langcode: en
```
Comment (swirtSJW) [inline config/sync/migrate_plus.migration.facility_health_care_service.yml:2]: This is actually a mistake that we just discovered in several of our migrations.  langcode should be removed from here and added to the `process` section.

File: config/sync/migrate_plus.migration.facility_health_care_service.yml [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+  revision_log:
+    plugin: default_value
+    default_value: 'Imported facility health services.'
+  revision_timestamp:
+    plugin: callback
+    callable: time
+  revision_uid:
+    plugin: default_value
+    default_value: 1317
+  uid:
```
Comment (swirtSJW) [inline config/sync/migrate_plus.migration.facility_health_care_service.yml:79]:
```suggestion
  langcode:
    plugin: default_value
    default_value: en
  uid:
```

File: docroot/modules/custom/va_gov_migrate/data/VAMC_Facility_Health_Services_Wilkes-Barre.csv (hunk 1 above)
Comment (swirtSJW) [reply docroot/modules/custom/va_gov_migrate/data/VAMC_Facility_Health_Services_Wilkes-Barre.csv:1]: You need to remove this file from your PR docroot/modules/custom/va_gov_migrate/data/VAMC_Facility_Health_Services_Wilkes-Barre.csv   
  We should not have two different CSVs in here.

File: config/sync/migrate_plus.migration.facility_health_care_service.yml [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+field_plugin_method: null
+cck_plugin_method: null
+migration_tags:
+  - Content
+  - Node
+migration_group: vamc
+label: 'Node - VAMC Facility Health Service.'
+source:
+  plugin: csv
+  path: modules/custom/va_gov_migrate/data/VAMC_Facility_Health_Services.csv
```
Comment (swirtSJW) [inline config/sync/migrate_plus.migration.facility_health_care_service.yml:15]: Please follow the pattern of existing CSVs in the data directory.  Use hyphens and no caps.
We already have the pattern of `regional-health-care-service.csv` so lets go with `facility-health-care-service.csv`

Comment (sundarka1) [reply config/sync/migrate_plus.migration.facility_health_care_service.yml:15]: I'll commit with these changes.

**Comment by swirtSJW** [comment]
I am seeing a bunch of these notices resulting from running this migration
```
#1 /var/aegir/projects/cms/pr2288/docroot/modules/contrib/tablefield/src/Plugin/Field/FieldFormatter/TablefieldFormatter.php(145): _drupal_error_handler(8, 'Undefined index...', '/var/aegir/proj...', 145, Array)
#2 /var/aegir/projects/cms/pr2288/docroot/core/lib/Drupal/Core/Field/FormatterBase.php(89): Drupal\tablefield\Plugin\Field\FieldFormatter\TablefieldFormatter->viewElements(Object(Drupal\Core\Field\FieldItemList), 'en')
#3 /var/aegir/projects/cms/pr2288/docroot/core/lib/Drupal/Core/Entity/Entity/EntityViewDisplay.php(263): Drupal\Core\Field\FormatterBase->view(Object(Drupal\Core\Field\FieldItemList), 'en')
#4 /var/aegir/projects/cms/pr2288/docroot/core/modules/layout_builder/src/Entity/LayoutBuilderEntityViewDisplay.php(267): Drupal\Core\Entity\Entity\EntityViewDisplay->buildMultiple(Array)
#5 /var/aegir/projects/cms/pr2288/docroot/core/lib/Drupal/Core/Entity/EntityViewBuilder.php(351): Drupal\layout_builder\Entity\LayoutBuilderEntityViewDisplay->buildMultiple(Array)
#6 /var/aegir/projects/cms/pr2288/docroot/core/modules/node/src/NodeViewBuilder.php(24): Drupal\Core\Entity\EntityViewBuilder->buildComponents(Array, Array, Array, 'default')
#7 /var/aegir/projects/cms/pr2288/docroot/core/lib/Drupal/Core/Entity/EntityViewBuilder.php(293): Drupal\node\NodeViewBuilder->buildComponents(Array, Array, Array, 'default')
#8 /var/aegir/projects/cms/pr2288/docroot/core/lib/Drupal/Core/Entity/EntityViewBuilder.php(250): Drupal\Core\Entity\EntityViewBuilder->buildMultiple(Array)
#9 [internal function]: Drupal\Core\Entity\EntityViewBuilder->build(Array)
```

image


I am investigating this.

File: docroot/modules/custom/va_gov_migrate/data/facility-health-care-service.csv [hunk 6]
```diff
@@ -0,0 +1,92 @@
+Id;Facility;VAMC System Health Service;Facility description of services;Owner
+'VAMC-FY-1';'Allentown VA Clinic';'Audiology and speech at VA Wilkes-Barre health care';'<h3>Location and contact information</h3><p class=""va-address-block""><strong></strong><br />3110 Hamilton Boulevard, <br />Allentown, PA 18103-3630<br /><strong>Phone:</strong> <a href=""tel: +1610-776-4304"">610-776-4304</a><br /><strong>Hours:</strong> Monday: 8:00 a.m.to 8:00 p.m. ET, <br />Tuesday through Friday: 8:00 a.m. to 4:30 a.m. ET</p><h3>Appointments</h3><p>Contact us to schedule, reschedule, or cancel your appointment.  If a referral is required, you will need to contact your primary care provider first.</p><p><strong>Referral needed? </strong> N<br /><strong>Walk-ins accepted?</strong> N<br /><strong>Phone: </strong><a href=""tel:+1610-776-4304"">610-776-4304</a> </p>';'VA Wilkes-Barre health care'
```
Comment (swirtSJW) [inline docroot/modules/custom/va_gov_migrate/data/facility-health-care-service.csv:2]: Phone numbers are not correct.  In your source you have them all double wrapped in quotes.  That is not valid html so the rendering of it is broken 
```
 <a href=""tel:+1570-759-0351"">
```
Should be 
```
 href="tel:+1570-759-0351">
```

The way it is now it ends up with broken html that looks like 
```
<a>570-824-3521</a>, ext. 24927
```
image

**Comment by swirtSJW** [comment]
@sundarka1 @SgardnerPMP This has been merged.  It will go out on on Monday's CMS deploy.  After that, let me know when you are ready for me to run the migration.

**Pull request merged**